packages/protocol/contracts/identity/FederatedAttestations.sol

```diff
@@ -0,0 +1,160 @@
+pragma solidity ^0.5.13;
+
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+import "openzeppelin-solidity/contracts/ownership/Ownable.sol";
+import "openzeppelin-solidity/contracts/token/ERC20/IERC20.sol";
+import "openzeppelin-solidity/contracts/utils/SafeCast.sol";
+
+import "./interfaces/IFederatedAttestations.sol";
+import "../common/interfaces/IAccounts.sol";
+import "../common/interfaces/ICeloVersionedContract.sol";
+
+import "../common/Initializable.sol";
+import "../common/UsingRegistry.sol";
+import "../common/Signatures.sol";
+import "../common/UsingPrecompiles.sol";
+import "../common/libraries/ReentrancyGuard.sol";
+
+/**
+ * @title Contract mapping identifiers to accounts
+ */
+contract FederatedAttestations is
+  IFederatedAttestations,
+  ICeloVersionedContract,
+  Ownable,
+  Initializable,
+  UsingRegistry,
+  ReentrancyGuard,
+  UsingPrecompiles
+{
+  using SafeMath for uint256;
+  using SafeCast for uint256;
+
+  struct Attestation {
+    address account;
+    uint256 issuedOn;
+    address signer;
+  }
+  // identifier -> issuer -> attestations
+  mapping(bytes32 => mapping(address => Attestation[])) public identifierToAddresses;
+  // account -> issuer -> identifiers
+  mapping(address => mapping(address => bytes32[])) public addressToIdentifiers;
+  // signer => revocation time
+  mapping(address => uint256) public revokedSigners;
+
+  // TODO ASv2 Event declarations
+
+  /**
+   * @notice Sets initialized == true on implementation contracts
+   * @param test Set to true to skip implementation initialization
+   */
+  constructor(bool test) public Initializable(test) {}
+
+  /**
+   * @notice Used in place of the constructor to allow the contract to be upgradable via proxy.
+   * @param registryAddress The address of the registry core smart contract.
+   */
+  function initialize(address registryAddress) external initializer {
+    _transferOwnership(msg.sender);
+    setRegistry(registryAddress);
+    // TODO ASv2 initialize any other variables here
+  }
+
+  /**
+   * @notice Returns the storage, major, minor, and patch version of the contract.
+   * @return The storage, major, minor, and patch version of the contract.
+   */
+  function getVersionNumber() external pure returns (uint256, uint256, uint256, uint256) {
+    return (1, 1, 0, 0);
+  }
+
+  function _isRevoked(address signer, uint256 time) internal returns (bool) {
```

Comment:
any reason this needs to be internal?

Reply:
It seems best practice with solidity function visibility is to be as conservative as possible and I couldn't think of a use case where this would need to be called outside of this contract. Happy to change it though

Reply:
yup sorry my comment was misguided -- was running into compile errors but needed to declare this as an internal view function (already the case in the compile fixes)